forte/data/ontology/top.py

```diff
@@ -686,5 +690,276 @@ def get(
         yield from self.pack.get(entry_type, self, components, include_sub_type)
 
 
+class ImageAnnotation(Entry):
+    def __init__(self, pack: PackType, image_payload_idx: int):
+        """
+        ImageAnnotation type entries, such as "edge" and "bounding box".
+        Each ImageAnnotation has a ``image_payload_idx`` corresponding to its
+        image representation in the payload array.
+
+        Args:
+            pack: The container that this image annotation
+                will be added to.
+            image_payload_idx: A integer that represents the index of
+                the image in the payloads.
+        """
+        super().__init__(pack)
+        self._image_payload_idx = image_payload_idx
+
+    @property
+    def image_payload_idx(self) -> int:
+        return self._image_payload_idx
+
+    @property
+    def image(self):
+        if self.pack is None:
+            raise ValueError(
+                "Cannot get image because image annotation is not "
+                "attached to any data pack."
+            )
+        return self.pack.get_image_array(self._image_payload_idx)
+
+    @property
+    def max_x(self):
+        return self.image.shape[1] - 1
+
+    @property
+    def max_y(self):
+        return self.image.shape[0] - 1
+
+    def __eq__(self, other):
+        if other is None:
+            return False
+        return self.image_payload_idx == other.image_payload_idx
+
+
+class Grids(Entry):
+    """
+    Regular grids with a grid configuration.
+
+    Args:
+        pack: The container that this grids will be added to.
+        height: the number of grid cell per column.
+        width: the number of grid cell per row.
+        image_payload_idx: the index of image in the datapack payloads.
+    """
+
+    def __init__(
+        self,
+        pack: PackType,
+        height: int,
+        width: int,
+        image_payload_idx: Optional[int] = None,
+    ):
+        super().__init__(pack)
+        if height <= 0 or width <= 0:
+            raise ValueError(
+                f"height({height}) and "
+                f"width({width}) both must be larger than 0"
+            )
+        self._height = height
+        self._width = width
+        if image_payload_idx is None:
+            self._image_payload_idx = 0
+        else:
+            self._image_payload_idx = image_payload_idx
+
+        self.img_arr = self.pack.get_image_array(self._image_payload_idx)
+        self.c_h, self.c_w = (
+            self.img_arr.shape[0] // self._height,
+            self.img_arr.shape[1] // self._width,
+        )  # compute the height and width of grid cells
+
+    def get_grid_cell(self, h_idx: int, w_idx: int):
```

Comment:
Need to define the convention on the meaning, the direction of `x`, `y`, is it counted from the top left corner, or from the bottom left corner.

Reply:
defined.